src/oneD/IonFlow.cpp

```diff
@@ -0,0 +1,416 @@
+//! @file IonFlow.cpp
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at http://www.cantera.org/license.txt for license and copyright information.
+
+#include "cantera/oneD/IonFlow.h"
+#include "cantera/oneD/StFlow.h"
+#include "cantera/base/ctml.h"
+#include "cantera/transport/TransportBase.h"
+#include "cantera/numerics/funcs.h"
+
+using namespace std;
+
+namespace Cantera
+{
+
+IonFlow::IonFlow(IdealGasPhase* ph, size_t nsp, size_t points) :
+    FreeFlame(ph, nsp, points),
+    m_phase(1),
+    m_inletVoltage(0),
+    m_outletVoltage(0),
+    m_kElectron(Undef)
+{
+    // make a local copy of species charge
+    for (size_t k = 0; k < m_nsp; k++) {
+        m_speciesCharge.push_back(m_thermo->charge(k));
+    }
+
+    // Find indices for charge of species
+    for (size_t k = 0; k < m_nsp; k++){
+        if (m_speciesCharge[k] != 0){
+            m_kCharge.push_back(k);
+        } else {
+            m_kNeutral.push_back(k);
+        }
+    }
+
+    // Find the index of electron 
+    if (m_thermo->speciesIndex("E") != npos ) {
+        m_kElectron = m_thermo->speciesIndex("E");
+    }
+
+    // mass fraction bounds (strict bound for ions)
+    for (size_t k : m_kCharge) {
+        setBounds(c_offset_Y+k, -1.0e-20, 1e-5);
+    }
+    setBounds(c_offset_P, -1.0e20, 1.0e20);
+    m_refiner->setActive(c_offset_P, false);
+
+    m_mobi.resize(m_nsp*m_points);
+    m_do_poisson.resize(m_points,false);
+    m_do_velocity.resize(m_points,true);
+}
+
+void IonFlow::resize(size_t components, size_t points){
+    StFlow::resize(components, points);
+    m_mobi.resize(m_nsp*m_points);
+    m_do_species.resize(m_nsp,true);
+    m_do_poisson.resize(m_points,false);
+    m_do_velocity.resize(m_points,true);
+    m_fixedElecPoten.resize(m_points,0.0);
+    m_fixedVelocity.resize(m_points);
+}
+
+void IonFlow::updateTransport(doublereal* x, size_t j0, size_t j1)
+{
+    StFlow::updateTransport(x,j0,j1);
+    for (size_t j = j0; j < j1; j++) {
+        setGasAtMidpoint(x,j);
+        m_trans->getMobilities(&m_mobi[j*m_nsp]);
+        if (m_kElectron != Undef) {
+            m_mobi[m_kElectron+m_nsp*j] = 0.4;
```

Comment:
The value came from "Numerical modelling of ion transport in flames" by J. Han et al. There are actually other approximations, but it will be more precise if it can be calculated by the Boltzmann's equation, which I am now working on. I should also make an option for users to choose.